include/sound/sof/ipc4/header.h

```diff
@@ -0,0 +1,313 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2021 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __INCLUDE_SOUND_SOF_IPC4_HEADER_H__
+#define __INCLUDE_SOUND_SOF_IPC4_HEADER_H__
+
+#include <linux/types.h>
+#include <uapi/sound/sof/abi.h>
+
+/** \addtogroup sof_uapi uAPI
+ *  SOF uAPI specification.
+ *  @{
+ */
+
+/*
+ * IPC4 messages have two 32 bit identifier made up as follows :-
+ *
+ * header - msg type, msg id, msg direction ...
+ * extension - extra params such as msg data size in mailbox
+ *
+ * These are sent at the start of the IPC message in the mailbox. Messages
+ * should not be sent in the doorbell (special exceptions for firmware).
+ */
+
+enum sof_ipc4_msg_target {
+	/* Global FW message */
+	SOF_IPC4_FW_GEN_MSG = 0,
+
+	/* Module message */
+	SOF_IPC4_MODULE_MSG = 1
+};
+
+enum sof_ipc4_global_msg {
+	SOF_IPC4_GLB_BOOT_CONFIG = 0,
+	SOF_IPC4_GLB_ROM_CONTROL = 1,
+	SOF_IPC4_GLB_IPCGATEWAY_CMD = 2,
+
+	SOF_IPC4_GLB_PERF_MEASUREMENTS_CMD = 13,
+	SOF_IPC4_GLB_CHAIN_DMA = 14,
+
+	SOF_IPC4_GLB_LOAD_MULTIPLE_MODULES = 15,
+	SOF_IPC4_GLB_UNLOAD_MULTIPLE_MODULES = 16,
+
+	/* pipeline settings */
+	SOF_IPC4_GLB_CREATE_PIPELINE = 17,
+	SOF_IPC4_GLB_DELETE_PIPELINE = 18,
+	SOF_IPC4_GLB_SET_PIPELINE_STATE = 19,
+	SOF_IPC4_GLB_GET_PIPELINE_STATE = 20,
+	SOF_IPC4_GLB_GET_PIPELINE_CONTEXT_SIZE = 21,
+	SOF_IPC4_GLB_SAVE_PIPELINE = 22,
+	SOF_IPC4_GLB_RESTORE_PIPELINE = 23,
+
+	/* Loads library (using Code Load or HD/A Host Output DMA) */
+	SOF_IPC4_GLB_LOAD_LIBRARY = 24,
+	SOF_IPC4_GLB_INTERNAL_MESSAGE = 26,
+
+	/* Notification (FW to SW driver) */
+	SOF_IPC4_GLB_NOTIFICATION = 27,
+	SOF_IPC4_GLB_MAX_IXC_MESSAGE_TYPE = 31
+};
+
+/* Message direction */
+enum sof_ipc4_msg_dir {
+	SOF_IPC4_MSG_REQUEST = 0,
+	SOF_IPC4_MSG_REPLY = 1,
+};
+
+enum sof_ipc4_pipe_line_state {
+	SOF_IPC4_PIPE_INVALID_STATE = 0,
+	SOF_IPC4_PIPE_UNINITIALIZED = 1,
+	SOF_IPC4_PIPE_RESET = 2,
+	SOF_IPC4_PIPE_PAUSED = 3,
+	SOF_IPC4_PIPE_RUNNING = 4,
+	SOF_IPC4_PIPE_EOS = 5
```

Comment:
ditto - if this enum has to fit into a fixed size.

Reply:
for enums, please do not initialize each item if they are incrementing by one.

Reply:
> for enums, please do not initialize each item if they are incrementing by one.

Done, thanks.